### ISSUE TITLE
Connection via Jumphost

### ISSUE THREAD

**Issue by joerg** [h3imdall/ftp-remote-edit#264]
Hi,

At my current work and many others Jumphosts are used to connect to servers. This means that we have to connect to the actual server via another SSH server. Currently I use local tunnels to connect with ftp-remote-edit to these servers, but it would be great to have this feature directly in Atom.

regards,
    Jörg

**Comment by srgk26** [comment]
Hi @joerg, I'm not sure that this issue would be resolved, but in the meantime, I'm trying to connect through Atom's ftp-remote-edit, but not sure how to do this. I know how to setup a local port forwarding, perhaps say like this:

`ssh -NfL 8888:serverB:22 serverB`

I've already configured server B to use server A as proxy jump host in my ~/.ssh/config file, so acts like: localhost --> server A --> server B

But now how would I go about setting up my atom's ftp-remote-edit config in atom to exploit this?

**Comment by nwzy** [comment]
I'm also interested in learning how @joerg accomplished this

I'm wondering if a possible strategy for ftp-remote-edit doing this is through the ssh config file

**Comment by joerg** [comment]
This is a feature request for a piece of software, not a common SSH support forum. For questions regarding SSH please consult Goole or Stackoverflow where these questions have been answered already numerous times.
Please do not misuse this ticket for anything else but the feature request that it is.

**Comment by srgk26** [comment]
Sorry if you felt I misused this platform @joerg, but this is not a common SSH question, I think it should absolutely be discussed here. I can quite easily navigate through my jump hosts and set up tunneling form the terminal, and even open up SFTP from my nautilus file manager, set up proxy config in my ~/.ssh/config folder, etc. The problem was how to accomplish this specifically for this ftp-remote-edit feature, like what you typed in this package's config folder in such a way that it respects the ssh config folder I've created in my ~/.ssh folder. I can easily accomplish this with vscode which follows my ssh configuration file. But it wasn't as easy to do this with this particular feature.

**Comment by julibeg** [comment]
It is indeed not trivial to find a configuration with jumping server that works with the mask used by ftp-remote-edit. I set up my `.ssh/config` so that I can use `ssh user@host` to connect through a jumping server onto my final remote. However, providing `user` as username and `host` as hostname for sftp in ftp-remote-edit doesn't work. 
Did you find a solution in the end, @srgk26?

**Comment by srgk26** [comment]
@julibeg No, never did. Also, I ended up migrating to vscode a few months ago actually (mainly cos the official Julia extension has shifted efforts from Atom to vscode). It was weird to adjust to vscode at first, but otherwise works great. At least it respects my ssh config.

**Comment by julibeg** [comment]
I see, thanks!

**Comment by julibeg** [comment]
For posterity, I got it working by using the default ssh port (`22`) on the remote (I wanted to avoid using `22` initially for reasons that go beyond this post). 
If you connect to your remote via jump server with the command `ssh bla` or `ssh user@bla`, then type `ssh -NL localhost:2222:localhost:22 bla` into a local terminal (instead of `2222` you can use any other port on your local machine, but you need to stick to `22` as port on the remote). Then, open the ftp-remote-edit server settings in atom, leave hostname as `localhost` and change the port to `2222` (or the local port you chose above). For protocol you of course pick SFTP and logon type SSH Agent. Then, provide your username on the remote (even if you already have the username in `.ssh/config`, you still need it here) and it should work.

**Comment by olayinkaajayi** [comment]
@julibeg Thank you for your suggestion. But I am unable to achieve the connection using the approach proposed. I keep getting the error message: "All configured authentication methods failed".

Can anyone still verify if this works.

**Comment by julibeg** [comment]
It still works for me. How is your jumphost setup? I recommend setting up your `~/.ssh/config` so that you can connect to the remote with `ssh myUserName@remoteName`. Then, you can open the tunnel with `ssh -NL localhost:2222:localhost:22 myUserName@remoteName` and add the connection to your ftp-remote-edit server list as described above. While the tunnel is open, you should then be able to connect. Hope that helps!

**Comment by olayinkaajayi** [comment]
Hello @julibeg , I was able to figure it out. I made use of `ssh user@jumphost -L 8888:remote_ip:12 -N `. Then I opened ftp-remote-edit on Atom and edited the configuration of the server using the username/passwrd option for logon type. The other information was as you suggested. With this, I successfully connected to the remote server whenever I clicked on the saved server name.
